lib/concurrent-ruby/concurrent/executor/ruby_thread_pool_executor.rb

```diff
@@ -114,9 +120,9 @@ def worker_task_completed
       synchronize { @completed_task_count += 1 }
     end
 
-    # @!macro thread_pool_executor_method_prune_pool
-    def prune_pool
```

Comment:
This public API no longer makes sense now that the pruning is automatic. It seems it was only [added](https://github.com/ruby-concurrency/concurrent-ruby/pull/938) because of the Ruby thread pool's synchronous pruning, which the Java version didn't suffer from, hence why it's a no-op there.

Reply:
Removing it seems too much of a breaking change though, how about keeping it as noop, maybe with a deprecation warning?

Reply:
Put it back and add a deprecation warning for this and the Java implementation.